### PR TITLE
feat: add support for long-running operations with rest transport.

### PR DESCRIPTION
**Pull request by kbandes**
This PR adds support for long-running operations in the rest transport.  Some noteworthy points:

- the python gapic generator now reads the service yaml file for an api, using a new `service_yaml` parameter in `py_gapic_library` targets.  The content of this file is loaded into a `google.api.Service` proto and is available in the schema as the `API.service_yaml_config` property.  This can, in the future, support other features that rely on information in this file.  Note that the java, golang, php, and nodejs targets already have this parameter.

- In addition to unit tests, this change has been tested against a live server using the cloud translate api.
Note that this change is dependent upon https://github.com/googleapis/python-api-core/pull/311.

### CODE REVIEW

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 1]
```diff
@@ -8,8 +8,10 @@ from google.api_core import retry as retries
 from google.api_core import rest_helpers
 from google.api_core import path_template
 from google.api_core import gapic_v1
+from google.api_core import operations_v1
```
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:11]:
```suggestion
```
This import exists 2 lines down.

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 2]
```diff
@@ -279,10 +287,16 @@ class {{service.name}}RestTransport({{service.name}}Transport):
...
+        {% if method.lro %}
+        return_op = operations_pb2.Operation()
+        json_format.Parse(response.content, return_op, ignore_unknown_fields=True)
+        return return_op
+        {% else %}
         return {{method.output.ident}}.from_json(
             response.content,
             ignore_unknown_fields=True
         )
+    {% endif %}
```
Comment (parthea) [inline gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:299]:
```suggestion
        {% endif %}
```

**Review by parthea — COMMENTED** [review]
Added minor observations

File: gapic/schema/api.py [hunk 3]
```diff
@@ -374,6 +384,23 @@ def services(self) -> Mapping[str, wrappers.Service]:
...
+        def make_http_options(rule: http_pb2.HttpRule
+                              ) -> Sequence[wrappers.HttpRule]:
+            http_options = [rule] + list(rule.additional_bindings)
+            opt_gen = (wrappers.HttpRule.try_parse_http_rule(http_rule)
+                       for http_rule in http_options)
+            return [rule for rule in opt_gen if rule]
+
+        result = {}
+        for rule in self.service_yaml_config.http.rules:
+            result[rule.selector] = make_http_options(rule)
```
Comment (software-dov) [inline gapic/schema/api.py:400]: Nit: we can do 
```python
result = {
    rule.selector: make_http_options(rule)
    for rule in self.service_yaml_config.http_rules
}
```

File: gapic/utils/options.py [hunk 4]
```diff
@@ -150,7 +164,8 @@ def tweak_path(p):
             add_iam_methods=bool(opts.pop('add-iam-methods', False)),
             metadata=bool(opts.pop('metadata', False)),
             # transport should include desired transports delimited by '+', e.g. transport='grpc+rest'
-            transport=opts.pop('transport', ['grpc'])[0].split('+')
+            transport=opts.pop('transport', ['grpc'])[0].split('+'),
+            service_yaml_config=service_yaml_config
```
Comment (software-dov) [inline gapic/utils/options.py:168]: Nit: please add a trailing comma

**Review by software-dov — APPROVED** [review]

**Comment by software-dov** [comment]
Looks good, just two minor nits that shouldn't block the merge. Feel free to integrate them or ignore them.

**Comment by kbandes** [comment]
I'll take a look.  In some cases the only way I could resolve conflicts between black and flake8 was to remove trailing commas, but that was a different repo.  I'll try it.

**Comment by kbandes** [comment]
I need to make additional changes:
1.  OperationsRestClient has been renamed to AbstractOperationsClient.
2.  The generated tests for rest now fail, I think because the Operations rest transport is not being mocked.

**Pull request merged**